annotation-interface/components/BrickForm.tsx

```diff
@@ -0,0 +1,176 @@
+import { UploadIcon } from '@heroicons/react/outline';
+import { XIcon, TrashIcon } from '@heroicons/react/solid';
+import { useRouter } from 'next/router';
+import { useEffect, useRef, useState } from 'react';
+
+import {
+    BrickUsage,
+    SupportedLanguage,
+    supportedLanguages,
+} from '../common/constants';
+import {
+    ITextBrick,
+    ITextBrickTranslation,
+    translationIsValid,
+    translationsToArray,
+    translationsToMap,
+} from '../database/models/TextBrick';
+import SelectionPopover from './SelectionPopover';
+const BrickForm = ({
```

Comment:
You might be able to utilize a Form library here to make your life much easier. I can highly recommend [React Hook Form](https://react-hook-form.com/).

React Hook Form could definitely replace most of your state and also validators (`useEffect`s)